common-pipeline/datarecord-metadata/datarecord-metadata-v3-schema.json

```diff
@@ -166,9 +169,40 @@
         "runtime_type": {
           "description": "Type of runtime",
           "type": "string"
+        },
+        "fraction_digits": {
+          "description": "Microsecond length for time and timestamp",
+          "type": "number"
+        },
+        "is_key": {
+          "description": "A field is key",
+          "type": "boolean"
+        },
+        "is_signed": {
+          "description": "Signed or Unsigned for number (bigint, integer, smallint, tinyint)",
+          "type": "boolean"
+        },
+        "item_index": {
```

Comment:
Can you give an example for this?  I'm not sure how this is used.

Reply:
Matt, I think you are asking for item_index, right?

The name of item_index is suggested by Curtis. 

If the row you are editing represents a column which is part of a subrecord, the item_index indicates the level of the column within the subrecord structure.

One example, Make Subrecord makes one column into multiple sub records. item_index could see which one is parent, which ones are sub records.

Reply:
Do you have an actual example?  I'm not following how this works.

Reply:
Yes, I am using Compare stage as example,
Compare stage compares two sets of input columns: column by column, and outputs the compare result column by column. As we cannot specify two groups of identical column names, and so we make them subrecords to give them unique names such as first.col1 and second.col1

![image](https://user-images.githubusercontent.com/64796288/81108952-357ccd00-8ece-11ea-9515-63cb2af4bc4b.png)

Reply:
This could also be accomplished via a disambiguation prefix on the property names. 

For example:

- DSLink4.ACCOUNT
- DSLink8.ACCOUNT

The prefix can be stored instead of an array index.

Reply:
In theory you could have nested sub-record infinitive. item_index is more flexible.

Reply:
It's still not clear to me how this is really used.  I can see how `source_column_id` is used even though I think that might/should change as well.  You have the with `source_column_id` which probably should be an object that has the `link_ref `and `field_name_ref` that describes where this field comes from.  Why do you need `item_index`?  Are you thinking field names aren't unique within this schema?  
`"Field name. Must be unique within the datarecord"`

Reply:
Matt,
1, item_index
Subrecord is widely used in restructure. 
Here is my input
![image](https://user-images.githubusercontent.com/64796288/81427189-4a946e80-910f-11ea-9f20-377adb7701e2.png)

I filter out **AutoPart** other than battery, then I make a subrecord output column batteryReadingsPerDay. This column has subrecords of input columns and also combine them into vector based on keys of ID and Date. 
Here is my output 
![image](https://user-images.githubusercontent.com/64796288/81429049-153d5000-9112-11ea-9c63-af85126168a4.png)

![image](https://user-images.githubusercontent.com/64796288/81429137-3736d280-9112-11ea-85d5-65f567de339f.png)

I found corresponding video which demos upper job. If you are interested, please check the session from 05:40 to 10:10.

2, source_column_id
It could be string with link_ref_id + field name.  If so, I this pipeline schema needs to support id for link_ref.

Reply:
Matt, did upper comments answer your question ? Thanks